### PR TITLE
feat: V-Text-Field no-resize

### PR DESCRIPTION
**Pull request by qm3ster**
## Description

Add the `resize: none` css attribute when a multiline input is `auto-grow`, or by manual `no-resize` attribute.
Also refactor all of the places that depend on the type of domnode to use a single `isTextarea` computed.
Also extract `row-height` as an attribute.
Also add validators to `row-height` and `rows` attributes.

## Motivation and Context

Manually resizing auto-grow inputs was undefined behavior and a bug.
Also visually removes the resize handle.

## How Has This Been Tested?

Added snapshot and unit tests.
Manual testing.

## Markup:

```vue
<template>
  <v-app>
    <v-content>
      <v-layout flex>
        <v-card>
          <v-text-field v-model='model' auto-grow multi-line rows='1'>

          </v-text-field>
        </v-card>
      </v-layout>
    </v-content>
  </v-app>
</template>

<script>
  export default {
    data: () => ({
      model: null
    })
  }
</script>
```
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

**Comment by qm3ster** [comment]
@johnleider there's a possible bug, please advise.
There are two sites (besides the classes) where a single one of the textarea-inducing props is used, rather than both props.

I am not sure if it is intentional in this case:
https://github.com/vuetifyjs/vuetify/blob/acaa6afb9e8ce2bbc95c16add041bd0fd6f7020f/src/components/VTextField/VTextField.js#L153

But I am almost **certain** it is a mistake here:
https://github.com/vuetifyjs/vuetify/blob/acaa6afb9e8ce2bbc95c16add041bd0fd6f7020f/src/components/VTextField/VTextField.js#L187-L192

Which of these, if any, care about the type of tag rather than the style?

**Comment by johnleider** [comment]
Can you move the row height to a prop please. Should accept [Number, String] with a default of 24.

**Comment by qm3ster** [comment]
How is this?

File: src/components/VTextField/VTextField.js [hunk 1]
```diff
@@ -6,6 +6,10 @@ import Input from '../../mixins/input'
 import Maskable from '../../mixins/maskable'
 import { isMaskDelimiter } from '../../util/mask'
 
+const warnNaN = (val, result) => {
```
Comment (johnleider) [inline src/components/VTextField/VTextField.js]: remove this

File: src/components/VTextField/VTextField.js [hunk 2]
```diff
@@ -35,9 +39,15 @@ export default {
     counter: [Boolean, Number, String],
     fullWidth: Boolean,
     multiLine: Boolean,
+    noResize: Boolean,
     placeholder: String,
     prefix: String,
+    rowHeight: {
```
Comment (johnleider) [inline src/components/VTextField/VTextField.js]: add validator

File: src/components/VTextField/VTextField.js [hunk 3]
```diff
@@ -35,9 +39,15 @@ export default {
...
+      type: [Number, String],
+      default: 24
+    },
     rows: {
```
Comment (johnleider) [inline src/components/VTextField/VTextField.js]: add validator

File: src/components/VTextField/VTextField.js [hunk 4]
```diff
@@ -103,8 +114,24 @@ export default {
...
+      return this.multiLine || this.textarea
+    },
+    noResizeHandle () {
+      return this.isTextarea && (this.noResize || this.autoGrow)
+    },
     shouldAutoGrow () {
-      return (this.multiLine || this.textarea) && this.autoGrow
+      return this.isTextarea && this.autoGrow
+    },
+    _rowHeight () {
```
Comment (johnleider) [inline src/components/VTextField/VTextField.js]: remove this, it's not needed

File: src/components/VTextField/VTextField.js [hunk 5]
```diff
@@ -103,8 +114,24 @@ export default {
...
+      const _rowHeight = parseFloat(this.rowHeight)
+      warnNaN(this.rowHeight, _rowHeight)
+      return _rowHeight
+    },
+    _rows () {
```
Comment (johnleider) [inline src/components/VTextField/VTextField.js]: remove this, it's not needed

File: src/components/VTextField/VTextField.js [hunk 6]
```diff
@@ -148,9 +175,8 @@ export default {
         const height = this.$refs.input
           ? this.$refs.input.scrollHeight
           : 0
-        const minHeight = this.rows * 24
-        const inputHeight = height < minHeight ? minHeight : height
-        this.inputHeight = inputHeight + (this.textarea ? 4 : 0)
+        const minHeight = this._rows * this._rowHeight
```
Comment (johnleider) [inline src/components/VTextField/VTextField.js]: just do `const minHeight = parseInt(this.rows) * parseInt(this.rowHeight)`

File: src/components/VTextField/VTextField.js [hunk 7]
```diff
@@ -235,10 +261,10 @@ export default {
 
       if (this.placeholder) data.domProps.placeholder = this.placeholder
 
-      if (!this.textarea && !this.multiLine) {
+      if (!this.isTextarea) {
         data.domProps.type = this.type
       } else {
-        data.domProps.rows = this.rows
+        data.domProps.rows = this._rows
```
Comment (johnleider) [inline src/components/VTextField/VTextField.js]: change to just rows, this isn't type dependent

**Comment by johnleider** [comment]
Bonus points for a unit test

File: src/components/VTextField/VTextField.js (hunk 6 above)
Comment (qm3ster) [reply src/components/VTextField/VTextField.js]: Wilco (and the above too)

File: src/components/VTextField/VTextField.js (hunk 7 above)
Comment (qm3ster) [reply src/components/VTextField/VTextField.js]: You mean, it should set `.rows` on an `input[type="text"]`?

**Comment by qm3ster** [comment]
By unit test you mean a snapshot test of the property set?

File: src/components/VTextField/VTextField.js [hunk 8]
```diff
@@ -44,11 +40,17 @@ export default {
...
+      default: 24,
+      validator (val) {
+        return !isNaN(parseFloat(val))
+      }
     },
     rows: {
       type: [Number, String],
-      default: 5
+      default: 5,
+      validator (val) {
```
Comment (johnleider) [inline src/components/VTextField/VTextField.js]: `validator: v => !isNaN(parseInt(val))`

File: src/components/VTextField/VTextField.js [hunk 9]
```diff
@@ -44,11 +40,17 @@ export default {
     prefix: String,
     rowHeight: {
       type: [Number, String],
-      default: 24
+      default: 24,
+      validator (val) {
```
Comment (johnleider) [inline src/components/VTextField/VTextField.js]: `validator: v => !isNaN(parseInt(val))`

**Comment by nekosaur** [comment]
Please update description of this PR with the following information

- bug fix or new feature
- description of what this PR does
- provide markup that shows the change being implemented
- does this PR required changes to documentation? and if so, what changes?

Also you need to make sure it passes CI

**Comment by qm3ster** [comment]
@nekosaur Updated description

**Comment by KaelWD** [comment]
Can you resolve conflicts and fix the lint errors?

**Comment by johnleider** [comment]
Closing due to activity

**Comment by qm3ster** [comment]
The adventure continues in #2994

**Pull request closed without merging**